azure-iot-device/azure/iot/device/provisioning/sk_registration_client.py

```diff
@@ -0,0 +1,80 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""
+This module contains one of the implementations of the Registration Client which uses Symmetric Key authentication.
+"""
+import logging
+from threading import Event
+from .registration_client import RegistrationClient
+from .models.registration_result import RegistrationResult
+from .internal.polling_machine import PollingMachine
+
+logger = logging.getLogger(__name__)
+
+
+class SymmetricKeyRegistrationClient(RegistrationClient):
+    """
+    Client which can be used to run the registration of a device using Symmetric Key authentication.
+    """
+
+    def __init__(self, mqtt_state_based_provider):
+        """
+        Initializer for the Symmetric Key Registration Client
```

Comment:
>Initializer for the Symmetric Key Registration Client [](start = 8, length = 53)

i'm all for docstrings but they should add something to the code. this has very little value for the user of the API

Reply:
The helpful statement is above the init. Are you in favor of deleting initializer docstrings and only having docstrings below the class name ?

---
In reply to: [277850407](https://github.com/Azure/azure-iot-sdk-python-preview/pull/81#discussion_r277850407) [](ancestors = 277850407)

Reply:
Initalizer docstrings are important, because we need to list the parameters for the initializer.

Class docstrings list the ivars and description for the class.

We need both. They are similar, but distinct.

---
In reply to: [277876199](https://github.com/Azure/azure-iot-sdk-python-preview/pull/81#discussion_r277876199) [](ancestors = 277876199,277850407)